torch/_tensor.py

```diff
@@ -30,12 +30,19 @@ def wrapped(*args, **kwargs):
             return NotImplemented
     return wrapped
 
-def _rebuild_from_type(func, type, args, dict):
-    if type is Tensor:
+def _rebuild_from_type(func, new_type, args, state):
+    if new_type is Tensor:
         return func(*args)
 
-    ret = func(*args).as_subclass(type)
-    ret.__dict__ = dict
+    ret = func(*args).as_subclass(new_type)
+    # Tensor does define __setstate__ even though it doesn't define
+    # __getstate__. So only use __setstate__ if it is NOT the one defined
+    # on Tensor
```

Comment:
Why does Tensor define setstate lol

Reply:
I don't know!
It even states there explicitely that `torch.load()` never uses it. And since we don't have a corresponding `__getstate__`, I don't know in which case that actually can be used properly...